src/Controller.js

```diff
@@ -0,0 +1,123 @@
+import FunctionTree from 'function-tree'
+import Module from './Module'
+import DefaultModel from './DefaultModel'
+import {ensurePath, isDeveloping, throwError} from './utils'
+import VerifyInputProvider from './providers/VerifyInput'
+import ContextProvider from 'function-tree/providers/Context'
+import StateProvider from './providers/State'
+import DebuggerProvider from './providers/Debugger'
+import ControllerProvider from './providers/Controller'
+import {EventEmitter} from 'events'
+import {dependencyStore as computedDependencyStore} from './Computed'
+
+/*
+  The controller is where everything is attached. The devtools
+  and router is attached directly. Also a top level module is created.
+  The controller creates the function tree that will run all signals,
+  based on top level providers and providers defined in modules
+*/
+class Controller extends EventEmitter {
+  constructor({state = {}, routes = {}, signals = {}, providers = [], modules = {}, router, devtools}) {
+    super()
+    this.flush = this.flush.bind(this)
+    this.devtools = devtools
+    this.model = new DefaultModel({})
```

Comment:
Good question. I initially started that path, but as I moved on I could not see any reason to attach other models. If it becomes necessary at a later point it is easy to add it. Let me just rename it to model.